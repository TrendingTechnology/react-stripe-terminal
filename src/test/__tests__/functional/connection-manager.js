import React from 'react';
import TestRenderer from 'react-test-renderer';
import { StripeTerminalProvider, injectStripeTerminal } from '../../../index';
import { TestSetup, TestComponent } from '../../helpers/test-setup';

describe('connectReader', () => {
    it('connects to a Reader and updates state', async () => {
        class TestClass extends TestComponent {
            render() {
                let { stripeTerminal } = this.props;
                return (
                    <div>
                        {stripeTerminal.connection.status}
                        {stripeTerminal.connection.reader
                            ? stripeTerminal.connection.reader.label
                            : null}
                    </div>
                );
            }
        }
        const TestApp = injectStripeTerminal(TestClass);
        const testElement = (
            <StripeTerminalProvider
                onFetchConnectionToken={TestSetup.FetchConnectionToken}
                onPaymentIntentRequest={() => {}}
                currency={'USD'}
            >
                <TestApp
                    stripeTerminalTestHook={t => {
                        stripeTerminal = t;
                    }}
                />
            </StripeTerminalProvider>
        );

        let stripeTerminal;
        const renderer = TestRenderer.create(testElement);
        let readers = await stripeTerminal.discoverReaders({
            method: 'simulated',
        });
        await stripeTerminal.connectReader(readers[0]);
        renderer.update(testElement);
        expect(renderer.toJSON()).toMatchSnapshot();
        await stripeTerminal.disconnectReader();
        renderer.update(testElement);
        expect(renderer.toJSON()).toMatchSnapshot();
    });
});
