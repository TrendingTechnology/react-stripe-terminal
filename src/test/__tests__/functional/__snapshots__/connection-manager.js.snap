// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectReader connects to a Reader and updates state 1`] = `
<div>
  connected
  SIMULATED
</div>
`;

exports[`connectReader connects to a Reader and updates state 2`] = `
<div>
  not_connected
</div>
`;
